SharedProjects/Babylon2GLTF/GLTFExporter.Skin.cs

```diff
@@ -42,110 +42,110 @@ private GLTFSkin ExportSkin(BabylonSkeleton babylonSkeleton, GLTF gltf, GLTFNode
-                    gltfBoneNode = _exportBone(babylonBone, gltf, babylonSkeleton, bones);
-                    babylonSkeletonExportData.nodeByBone.Add(babylonBone, gltfBoneNode);
-                }
-                gltfBoneNode = babylonSkeletonExportData.nodeByBone[babylonBone];
-
-                gltfJoints.Add(gltfBoneNode.index);
-
-                // Set this bone as skeleton if it is a root
-                // Meaning of 'skeleton' here is the top root bone
-                if (babylonBone.parentBoneIndex == -1)
-                {
-                    gltfSkin.skeleton = gltfBoneNode.index;
-                }
-
-                // Compute inverseBindMatrice for this bone when attached to this node
-                var boneLocalMatrix = new BabylonMatrix();
-                boneLocalMatrix.m = babylonBone.matrix;
-                //printMatrix("boneLocalMatrix[" + babylonBone.name + "]", boneLocalMatrix);
-
-                BabylonMatrix boneWorldMatrix = null;
-                if (babylonBone.parentBoneIndex == -1)
-                {
-                    boneWorldMatrix = boneLocalMatrix;
-                }
-                else
-                {
-                    var parentWorldMatrix = bonesWorldMatrices[babylonBone.parentBoneIndex];
-                    // Remove scale of parent
-                    // This actually enable to take into account the scale of the bones, except for the root one
-                    parentWorldMatrix = _removeScale(parentWorldMatrix);
-
-                    boneWorldMatrix = boneLocalMatrix * parentWorldMatrix;
-                }
-                //printMatrix("boneWorldMatrix[" + babylonBone.name + "]", boneWorldMatrix);
-
-                var inverseBindMatrices = nodeWorldMatrix * BabylonMatrix.Invert(boneWorldMatrix);
-
-                // Populate accessor
-                List<float> matrix = new List<float>(inverseBindMatrices.m);
-                matrix.ForEach(n => accessorInverseBindMatrices.bytesList.AddRange(BitConverter.GetBytes(n)));
-                accessorInverseBindMatrices.count++;
-            }
+            var babylonSkeletonExportData = alreadyExportedSkeletons[babylonSkeleton];
+
+            // Skin
+
+            // if this mesh is sharing a skin with another mesh, use the already exported skin
+            var sharedSkinnedMeshesByOriginalPair = sharedSkinnedMeshesByOriginal.Where(skinSharingMeshPair => skinSharingMeshPair.Value.Contains(gltfMesh)).Select(kvp => (KeyValuePair<GLTFMesh, List<GLTFMesh>>?) kvp).FirstOrDefault();
+            if (sharedSkinnedMeshesByOriginalPair != null)
+            {
+                logger.RaiseMessage("GLTFExporter.Skin | Sharing skinning information from mesh '" + sharedSkinnedMeshesByOriginalPair.Value.Key.name + "'", 3);
+                var skeletonExportData = alreadyExportedSkeletons[babylonSkeleton];
+                gltfNode.skin = skeletonExportData.skinIndex;
+                return gltf.SkinsList[(int)gltfNode.skin];
```

Comment:
is this change related to the goal of the PR?

Reply:
no this was a bug, introduced after the last refactoring I think

Reply:
thanks for the fix